src/stage_prof.cpp

```
@@ -0,0 +1,163 @@
+    c->thr_busy_min = c->thr_busy_max = c->thr_busy_mean = c->thr_busy_stdev = NAN;
+    c->write_compress = c->write_diskwrite = NAN;
+    c->chunk_start = NAN;
+    c->read_bytes_in = c->bgzf_blocks = -1;   /* -1 -> blank (N/A) */
+}
+
+void sp_add_chunk(const prof_chunk_t *c) {
+    if (!g_on) return;
+    pthread_mutex_lock(&g_mu);
+    if (g_n == g_cap) {
+        g_cap = g_cap ? g_cap * 2 : 64;
+        g_rows = (prof_chunk_t*)realloc(g_rows, g_cap * sizeof *g_rows);
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)